pkg/procsyswrapper/procsys.go

```diff
@@ -42,3 +45,8 @@ func (p *procSys) Get(key string) (string, error) {
 func (p *procSys) Set(key, value string) error {
 	return ioutil.WriteFile(p.path(key), []byte(value), 0644)
 }
+
+// IsPathWriteAccessible verifies if aws-node pod can write to the specified path
+func (p *procSys) IsPathWriteAccessible(key string) bool {
```

Comment:
This encourages TOCTOU (time of check is not time of use) race attacks.  A more robust pattern is to just attempt the real `Set` and see if the error result is "permission denied" (`os/error.IsPermission(err)` in golang).

Reply:
oops looks like I missed this :/ Here TOCTOU is not possible because when aws-node starts up it will either have /proc accessible or not depending on privileged flag. It won't change during the execution. Please correct me if otherwise.